aea/protocols/dialogue/base.py

```diff
@@ -1134,18 +1134,50 @@ def remove(self, dialogue_label: DialogueLabel) -> None:
 
         :param dialogue_label: label of the dialogue to remove
         """
-        dialogue = self._dialogues_by_dialogue_label.pop(dialogue_label, None)
+        dialogue_reference = dialogue_label.dialogue_reference
 
-        self._incomplete_to_complete_dialogue_labels.pop(dialogue_label, None)
+        # Check if dialogue_label is a complete label
+        is_complete_label = (
+            dialogue_reference[0] != Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+            and dialogue_reference[1] != Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+        )
 
-        if dialogue_label in self._terminal_state_dialogues_labels:
-            self._terminal_state_dialogues_labels.remove(dialogue_label)
+        complete_dialogue_label = dialogue_label
 
-        if dialogue:
-            self._dialogue_by_address[dialogue_label.dialogue_opponent_addr].remove(
-                dialogue
+        # Reconstruct and delete incomplete label
+        if is_complete_label:
+
+            incomplete_dialogue_label = complete_dialogue_label.get_incomplete_version()
+
+            self._incomplete_to_complete_dialogue_labels.pop(
+                incomplete_dialogue_label, None
             )
 
+            if incomplete_dialogue_label in self._terminal_state_dialogues_labels:
+                self._terminal_state_dialogues_labels.remove(incomplete_dialogue_label)
+
+            dialogue = self._dialogues_by_dialogue_label.pop(
+                incomplete_dialogue_label, None
+            )
+
+            if dialogue:
+                self._dialogue_by_address[
+                    incomplete_dialogue_label.dialogue_opponent_addr
+                ].remove(dialogue)
+
+        # Delete complete label
+        self._incomplete_to_complete_dialogue_labels.pop(complete_dialogue_label, None)
+
+        if complete_dialogue_label in self._terminal_state_dialogues_labels:
+            self._terminal_state_dialogues_labels.remove(complete_dialogue_label)
```

Comment:
Are we sure this `self._terminal_state_dialogues_labels` can never contain `incomplete` labels?